chia/consensus/blockchain.py

```diff
@@ -374,13 +378,7 @@ async def _reconsider_peak(
                     lastest_coin_state[coin_record.name] = coin_record
 
             # Rollback sub_epoch_summaries
```

Comment:
Shouldn't we roll back height_to_hash as well?

Reply:
There is a possibility that a reorg results in a peak with a lower height than the previous peak. Would that cause any issues here?

Reply:
I did that in an earlier version of this patch, but we have a test that fails when I did that. The existing behavior (without this patch) is that we don't roll back as well. I think it would probably make sense to understand why it fails (I suspect the test accidentally depends on the existing behavior). I don't think it should be a blocker for this PR